kvserver: minor test cleanup

andreimatei opened the pull request:
See individual commits.

andreimatei committed:
kvserver: minor test cleanup

This test was uselessly looking at the cancelation of an uncancellable
context.

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93363)
<!-- Reviewable:end -->

andreimatei committed:
kvserver: minor test simplification

This test was jumping through hoops in order to combine a cluster
stopper, with its own Tracer, with calling into a node, which has a
different Tracer. Normally combining Tracers like that is illegal, but
the test was making sure to force the Stopper to create "sterile" spans
for its tasks, which are exempt from this restriction about combining
them with other Tracers, for arcane reasons. This patch switches the
test to a more straight-forward hoop: don't use a context created by the
stopper when calling into the node. Besides being simpler, this is
needed cause the exception for sterile spans is going away.

Release note: None
Epic: None

andreimatei reviewed (commented):
friendly ping

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/93363#-:-NJCvMPY01C1HE2dXzqL:b-50swta)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)



<!-- Sent from Reviewable.io -->

andreimatei commented:
bors r+
